cpp/ps0005/main.cpp

```diff
@@ -4,47 +4,65 @@ using namespace std;
 
 int main()
 {
-    int a,b=0;
-    float c,d=0;
+    // prompting user for 2 operands and an operator 1.5/2
+    float num1, num2;
     char proc;
+    
     cout<<"please Enter Two number :)\nnow you can Enter frist number a= ";
-    cin>>a;
+    cin>>num1;
     if(cin.fail()==true)
     {
-    cout<<"Error please input number\n";
-    system("pause");
-    return 0;
+        cout<<"Error please input number\n";
+        system("pause");
+        return 5;
     }
-    else
+    
     cout<<"now you can Enter scond number b= ";
-    cin>>b;
+    cin>>num2;
     if(cin.fail()==true)
     {
-    cout<<"Error please input number\n";
-
-    system("pause");
-    return 0;
+        cout<<"Error please input number\n";
+        system("pause");
+        return 5;
     }
-    else
+    
+    int a=num1, b=num2;
+    
     cout<<"Choose operation (+,-,*,/): \n";
     cin>>proc;
+
+    // checking user inputs and processing 2/4
+    // printing the result if all data is valid 1/1
     switch (proc)
     {
-    case '+':cout<<"("<<a<<"+"<<b<<")= "<<a+b<<endl;break;
-    case '-':cout<<"("<<a<<"-"<<b<<")= "<<a-b<<endl;break;
-    case '*':cout<<"("<<a<<"*"<<b<<")= "<<a*b<<endl;break;
-    case '/':{
-    if(b!=0)
-    {
-    c=(float)a;
-    d=(float)b;
-    cout<<"("<<c<<"/"<<d<<")= "<<c/d<<endl;
-                }
-    else
-    cout<<"Error"<<a<<"/0 Wrong\n";};break;
-    default:cout<<"Wrong choice you can choice between(-,+,*,/)\n";break;
-    }
-    system("pause");
-    return 0;
+        case '+':cout<<"("<<num1<<"+"<<num2<<")= "<<num1+num2<<endl;break;
+        case '-':cout<<"("<<num1<<"-"<<num2<<")= "<<num1-num2<<endl;break;
+        case '*':cout<<"("<<num1<<"*"<<num2<<")= "<<num1*num2<<endl;break;
+        
+        case '/':{
+            if(num2!=0)
+                cout<<"("<<num1<<"/"<<num2<<")= "<<num1/num2<<endl;
+            else
+            {
+                if(num1!=0)
+                    cout<<"Error"<<num1<<"/0 Wrong\n";
+                else
+                    cout<<"Infinite number of solutions: (0 / 0) is Undefined Quantity.\n";
+            }
+        } // 0.5/1
+        break;
 
+        case '%':{
+            if(num1==a && num2==b)
+                cout<<"("<<num1<<"%"<<num2<<")= "<<a%b<<endl;
+            else
+                cout<<"Cannot execute modulo operation for floating point numbers.\n";
+        } // 0/1
+        break;
+        
+        default:cout<<"Wrong choice you can choice between(-,+,*,/)\n";break; // 1/1
+    } // 0.5/1
+    
+    return 0;
 }
+// good work 👍 7/10

```

Comment:
- [x] program is in the correct path
- [x] no syntax errors while compilation
- [ ] formatted code to be clear and readable
- [ ] no bugs or warnings while execution
- [ ] commented some phrases for blocks of code
- [ ] main returns non-zero/zero number according to the case
+1 bonus mark for your effort ⭐

you can check the [problem solution](https://onlinegdb.com/9tQHZBXgL) for more